src/lib/Libifl/pbsD_submit_resv.c

```diff
@@ -77,50 +77,16 @@ __pbs_submit_resv(int c, struct attropl *attrib, char *extend)
 	for (pal = attrib; pal; pal = pal->next) {
 		pal->op = SET;		/* force operator to SET */
 		if (msvr) {
-			if (strcmp(pal->name, ATTR_convert) == 0 || strcmp(pal->name, ATTR_resv_job) == 0) {
+			/* check if it is job-specific ASAP/Now reservation. In both the cases we have a job id */
+			if (strcmp(pal->name, ATTR_convert) == 0 || strcmp(pal->name, ATTR_resv_job) == 0)
 				strcpy(job_id_resv, pal->value);
-			}
 		}
 	}
 
 	if (msvr) {
-		/* Job specific reservation */
-		if (job_id_resv[0] != '\0') {
-			/* Do a job stat to find out the server_instance_fd of
-			* the server instance where the job resides
-			*/
-			struct attrl *attr;
-			struct batch_status *ss = NULL;
-			char *svr_inst_id = NULL;
-			static struct attrl attribs[] = {
-				{	NULL,
-					ATTR_server_inst_id,
-					NULL,
-					"",
-					SET
-				}	
-			};
-
-			ss = pbs_statjob(c, job_id_resv, attribs, NULL);
-			if (ss == NULL)
-				return NULL;
-
-			if (ss != NULL) {
-				for (attr = ss->attribs; attr != NULL; attr = attr->next) {
-					if (strcmp(attr->name, ATTR_server_inst_id) == 0) { 
-						svr_inst_id = strdup(attr->value);
-						if (svr_inst_id == NULL) {
-							pbs_statfree(ss);
-							return NULL;
-						}
-						break;
-					}
-				}
-			}    
-			c = get_svr_inst_fd(c, svr_inst_id);
-			free(svr_inst_id);
-			pbs_statfree(ss);
-		} else { 
+		if (job_id_resv[0] != '\0') /* Job specific reservation */
+			c = get_job_svr_inst_id(c, job_id_resv);
```

Comment:
I think we should rename this function to "stat_job_svr_inst_id" to make it clear that this will perform a stat IFL call to get the instance id

Reply:
By keeping the function name as get_job_svr_inst_id(), we are trying to hide the implementation details i.e. the user does not need to know that we are doing a job-stat to get the server instance id. Tomorrow we might choose some other logic to get server instance id. I feel this is a good abstraction/encapsulation. 

Just worried that if we change the name to "stat_job_svr_inst_id" the name of the function itself says that we are doing a job stat. So just wanted to leave this as is for now and if required we can always change the name whenever needed.